packages/vite/src/node/plugins/resolve.ts

```diff
@@ -813,6 +813,9 @@ export function tryNodeResolve(
     exclude?.includes(pkgId) ||
     exclude?.includes(nestedPath) ||
     SPECIAL_QUERY_RE.test(resolved) ||
+    // During dev SSR, we don't have a way to reload the module graph if
+    // a non-optimized dep is found. So we need to skip optimization here.
+    // The only optimized deps are the ones explicitly listed in the config.
     (!isBuild && ssr) ||
```

Comment:
```suggestion
    (!options.ssrOptimizeCheck && !isBuild && ssr) ||
```
Don't we need to add `!options.ssrOptimizeCheck` here?
IIUC `addManuallyIncludedOptimizeDeps` uses this too, so the explicitly listed ones won't be optimized.

Reply:
You are right, I don't understand how the code evolved into its current state 🤔 
This means that we didn't have any external tests for this feature for a while.